src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx

```diff
@@ -17,35 +17,39 @@ import { i18n } from '@osd/i18n';
 import { of } from 'rxjs';
 import { WorkspaceAttribute } from 'opensearch-dashboards/public';
 import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
-import { WorkspaceForm, WorkspaceFormData } from '../workspace_creator/workspace_form';
+import {
+  WorkspaceForm,
+  WorkspaceFormSubmitData,
+  WorkspaceFormData,
+} from '../workspace_creator/workspace_form';
 import { WORKSPACE_OVERVIEW_APP_ID, WORKSPACE_OP_TYPE_UPDATE } from '../../../common/constants';
 import { DeleteWorkspaceModal } from '../delete_workspace_modal';
 import { formatUrlWithWorkspaceId } from '../../utils';
 import { WorkspaceClient } from '../../workspace_client';
 
+function getFormDataFromWorkspace(
+  currentWorkspace: WorkspaceAttribute | null | undefined
+): WorkspaceFormData {
+  currentWorkspace = currentWorkspace || ({} as WorkspaceAttribute);
+  return { ...currentWorkspace, permissions: [] };
+}
 export const WorkspaceUpdater = () => {
   const {
     services: { application, workspaces, notifications, http, workspaceClient },
   } = useOpenSearchDashboards<{ workspaceClient: WorkspaceClient }>();
 
   const currentWorkspace = useObservable(workspaces ? workspaces.currentWorkspace$ : of(null));
-
-  const excludedAttribute = 'id';
-  const { [excludedAttribute]: removedProperty, ...otherAttributes } =
-    currentWorkspace || ({} as WorkspaceAttribute);
-
   const [deleteWorkspaceModalVisible, setDeleteWorkspaceModalVisible] = useState(false);
-  const [currentWorkspaceFormData, setCurrentWorkspaceFormData] = useState<
-    Omit<WorkspaceAttribute, 'id'>
-  >(otherAttributes);
+  const [currentWorkspaceFormData, setCurrentWorkspaceFormData] = useState<WorkspaceFormData>(
+    getFormDataFromWorkspace(currentWorkspace)
```

Comment:
The currentWorkspace will includes the permission data, we can't override with empty array.